pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -130,10 +142,17 @@ func (r *ReconcileBareMetalHost) Reconcile(request reconcile.Request) (reconcile
 			return reconcile.Result{}, nil
 		}
 
-		// NOTE(dhellmann): This is where we would do something with
-		// external resources not managed through CRs (those are
-		// deleted automatically), like telling ironic to wipe the
-		// host.
+		if dirty, err = r.provisioner.Remove(host); err != nil {
+			reqLogger.Error(err, "failed to deprovision")
+			return reconcile.Result{}, err
+		}
+		if dirty {
+			if err := r.saveStatus(host); err != nil {
+				reqLogger.Error(err, "failed to clear host status on deprovision")
+				return reconcile.Result{}, err
+			}
+			return reconcile.Result{Requeue: true}, nil
```

Comment:
I'm anticipating Remove() taking several steps, and us wanting to wait for it if we are wiping the server.

Reply:
I added a comment to explain this expectation.

Reply:
Gotcha. Probably then we'll want to use the `RequeueAfter` option so that there is some delay before reconcile happens again. Is there any way for the operator to estimate how long it will need to wait? If not, we can just select a sensible polling interval.